### ISSUE TITLE
About results

### ISSUE THREAD

**Issue by SakurajimaMaiii** [yueatsprograms/ttt_cifar_release#5]
Thanks for the code!
I try to re-implement TTT-online.
On the original testset of CIFAR10, I got 8.58 % error rate, under joint training setting.
**The improvements seem like less than results in paper.**
My results:
![TTT_results](https://user-images.githubusercontent.com/83657651/178941101-8415f40d-99ac-470f-a4bc-a232b39758a0.JPG)
the results in paper:
![捕获](https://user-images.githubusercontent.com/83657651/178942059-0961df2b-1249-4a98-9be6-e1203acb9e9c.png)

 I wonder if this result makes sense or not.
Environment:
	Python: 3.8.13
	PyTorch: 1.12.0+cu116
	Torchvision: 0.13.0+cu116
	CUDA: 11.6
	CUDNN: 8302
	NumPy: 1.22.3
	PIL: 9.0.1

Thanks

**Comment by yueatsprograms** [comment]
It seems like the baseline is much worse to start with? Also, did you re-implement or simply cloned this code and ran?

**Comment by SakurajimaMaiii** [comment]
I just clone and ran   (it seems like in utils/prepare_dataset.py, _import numpy as np_ is needed..)
It is strange:
under joint training.
When I use GN (group_norm=8), the baseline result is 8.58% (higher  than paper 8.1%)
use BN, the baseline result is 6.7% (lower in paper 7.5%)

**Comment by yueatsprograms** [comment]
I have reran this code a few times over the past two years on different machines, and I have been getting maybe +-0.1% for the GN model. Maybe try to run it with a different random seed? Beyond this I'm really not sure what the problem could be.

If you want to use the GN model for research, you could maybe just train it for more epochs, with the milestones changed proportionately.

**Comment by SakurajimaMaiii** [comment]
I think I know the reason. It's my mistake.
Thank you for your patience and help.

Best

**Comment by yueatsprograms** [comment]
Good to hear that you solved it. You are welcome.
